src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollectionItem.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CommunityToolkit.Diagnostics;
+using CommunityToolkit.Mvvm.ComponentModel;
+using Microsoft.Toolkit.Uwp.UI.Controls;
+using OwlCore.Extensions;
+using StrixMusic.Sdk.AppModels;
+using StrixMusic.Sdk.ViewModels;
+
+namespace StrixMusic.Shells.ZuneDesktop.Controls.Views.Collection
+{
+    /// <summary>
+    /// An container for items in a <see cref="ZuneTrackCollection"/>, with added functionality and observable properties.
+    /// </summary>
+    public sealed partial class ZuneTrackCollectionItem : ObservableObject
+    {
+        // Cache artist item values
+        // Keeping a minimum cache that updates with events allows us to
+        // avoid checking all items when a single item updates.
+        private ConcurrentDictionary<string, int> _lastKnownTrackArtistsCount = new();
+
+        [ObservableProperty]
+        private ITrackCollectionViewModel? _parentCollection;
+
+        [ObservableProperty]
+        private TrackViewModel? _track;
+
+        [ObservableProperty]
+        private bool _shouldShowArtistList;
+
+        [ObservableProperty]
+        private ObservableCollection<MetadataItem> _artistNamesMetadata = new ObservableCollection<MetadataItem>();
+
+        partial void OnTrackChanging(TrackViewModel? newValue)
+        {
+            _ = ExecuteAsync(newValue);
```

Comment:
Try not to make local methods if its the only thing your method is doing, and discard the async method.

Reply:
This method cannot return `Task` because it's implemented by the source generator. This is needed to properly fire and forget the task.